core/storage/file/gae_models_test.py

```diff
@@ -0,0 +1,229 @@
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file2.png')
+        file_metadata_model2.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model_list = (
+            file_models.FileMetadataModel.get_undeleted())
+        self.assertEqual(len(file_metadata_model_list), 2)
+        self.assertEqual(
+            file_metadata_model1, file_metadata_model_list[0])
+        self.assertEqual(
+            file_metadata_model2, file_metadata_model_list[1])
+
+    def test_get_undeleted_with_one_deleted_and_undeleted_model_returns_one(
+            self):
+        file_metadata_model1 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model1.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file2.png')
+        file_metadata_model2.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model2.delete(
+            feconf.SYSTEM_COMMITTER_ID, 'Delete second file model')
+        file_metadata_model_list = (
+            file_models.FileMetadataModel.get_undeleted())
+        self.assertEqual(len(file_metadata_model_list), 1)
+        self.assertEqual(
+            file_metadata_model1, file_metadata_model_list[0])
+
+    def test_create_creates_model_with_correct_id_and_deleted_status_as_false(
+            self):
+        file_metadata_model1 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        self.assertEqual(file_metadata_model1.deleted, False)
+        self.assertEqual(
+            file_metadata_model1.id, '/exp_id1/path/to/file1.png')
+
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', '/exp_id1/path/to/file2.png')
+        self.assertEqual(file_metadata_model2.deleted, False)
+        self.assertEqual(
+            file_metadata_model2.id, '/exp_id1/path/to/file2.png')
+
+    def test_get_model_with_model_present_returns_the_correct_model(self):
+        file_metadata_model = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        retrieved_model = file_models.FileMetadataModel.get_model(
+            'exp_id1', 'path/to/file1.png')
+
+        self.assertEqual(retrieved_model, file_metadata_model)
+
+    def test_get_model_non_strict_with_no_model_present_returns_none(self):
+        retrieved_model = file_models.FileMetadataModel.get_model(
+            'exp_id1', 'path/to/file2.png')
+        self.assertIsNone(retrieved_model)
+
+    def test_get_model_strict_with_no_model_present_raises_error(self):
+        with self.assertRaisesRegexp(
+            base_models.BaseModel.EntityNotFoundError, (
+                'Entity for class FileMetadataModel '
+                'with id /exp_id1/path/to/file2.png not found')):
```

Comment:
For here & other assertion errors, consider using regex to remove the part of the error message that relies on an ID. This would help make the tests less fragile/likely to break if the implementation changes drastically (e.g. if it changes how its key or ID naming strategy). For this error, maybe try something like:

``'Entity for class FileMetadataModel with id.+?not found'``